crates/router/src/routes/dummy_connector/utils.rs

```diff
@@ -0,0 +1,98 @@
+use app::AppState;
+use error_stack::{IntoReport, ResultExt};
+use masking::ExposeInterface;
+use rand::Rng;
+use redis_interface::RedisConnectionPool;
+use tokio::time as tokio;
+
+use super::{
+    errors::DummyConnectorErrors,
+    types::{
+        DummyConnectorPaymentData, DummyConnectorPaymentMethodData, DummyConnectorPaymentRequest,
+        DummyConnectorPaymentResponse, DummyConnectorResponse, DummyConnectorTransactionStatus,
+    },
+};
+use crate::{connection, core::errors, logger, routes::app, services::api};
+
+pub async fn tokio_mock_sleep(delay: u64, tolerance: u64) {
+    let mut rng = rand::thread_rng();
+    let add: bool = rng.gen();
+    let effective_delay = if add {
+        delay + rng.gen_range(0..tolerance)
+    } else {
+        delay - rng.gen_range(0..tolerance)
+    };
```

Comment:
can you please add the logic that we've discussed to remove generation of 2 random states